### ISSUE TITLE
Links don't open in browser

### ISSUE THREAD

**Issue by quesada** [moosotc/llpp#105]
Clicking links doesn't open them in browser (KDE; default set to vivaldi). Checked conf file, no reference to set default browser.

**Comment by moosotc** [comment]
Jose Quesada <notifications@github.com> writes:

> Clicking links doesn't open them in browser (KDE; default set to
> vivaldi). Checked conf file, no reference to set default browser.

`uri-launcher' is the knob you want to tweak. I'm not sure how it's
supposed to be done in KDE, but you can try setting it to
 `xdg-open %s'
for instance.

-- 
mailto:moosotc@gmail.com

**Comment by quesada** [comment]
Thanks. Great job with llpp

**Comment by zhenjieliang** [comment]
using `xdg-open %s' can open links  such as https://github.com or pdf_path such as file://local_files/test.pdf,

but can not open links to files such as file://local_files/test.tex

even when I changed the uri-laucher from xdg-open %s to nvim %s

still can not open links such as file://local_files/test.tex

**Comment by zhenjieliang** [comment]
Thanks! You are really a great developer and maintainer!

I compiled the tex file to get my pdf, in the tex file the excerpt is 

```
\ \href{run:/media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex}{elegantbook-cn.tex}。
```

so that the compiled pdf will show a link such as file:///media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex}{elegantbook-cn.tex}

still cant open the tex file.

**Comment by moosotc** [comment]
zhenjieliang <notifications@github.com> writes:

> Thanks! You are really a great developer and maintainer!
>
> I compiled the tex file to get my pdf, in the tex file the excerpt is 
>
> ```
> \
> \href{run:/media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex}{elegantbook-cn.tex}
> 。
> ```
>
> so that the compiled pdf will show a link such as
> file:///media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex}{elegantbook-cn.tex}
>
> still cant open the tex file.

The above url looks weird... the braces are unbalanced and look out of
place even if they were. And i do not know tex to be of any help sorry,
you would have to ask somebody else.

**Comment by zhenjieliang** [comment]
> The above url looks weird...



sorry, I typed the wrong links, the correct one was 
```
 file:///media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex
```

I have created a minimal example named llpp_link_minimal_tex.zip, please download and unzip it. the tex needed compiled is the minimal.tex

```
\documentclass{article}
\usepackage{hyperref}
\begin{document}

My file links are as follows

\begin{enumerate}
\item \href{run:local_files/copula.pdf}{relative path pdf link}
\item \href{run:/media/liangzhenjie/Desktop/notes/local_files/llpp_link_minimal_tex/minimal.pdf}{absolute path pdf link}
\item \href{run:local_files/test.tex}{relative path tex link}
\item \href{run:/media/liangzhenjie/Desktop/notes/local_files/llpp_link_minimal_tex/local_files/test.tex}{absolute path tex link}
\end{enumerate}

please change the absolute path in your system to make right. You will see that all tex links failed, while all pdf links worked.


\end{document}

```

The compiling command is `xelatex minimal.tex`, but before compiling, please change the absolute path of pdf and tex in it.

Then click the tex link in the output pdf, you will see that it failed to open the test.tex.







[llpp_link_minimal_tex.zip](https://github.com/moosotc/llpp/files/5106185/llpp_link_minimal_tex.zip)

**Comment by moosotc** [comment]
zhenjieliang <notifications@github.com> writes:

>> The above url looks weird...
>
>
>
> sorry, I typed the wrong links, the correct one was 
> ```
>  file:///media/liangzhenjie/Desktop/notes/local_files/my_materials/elegantbook-cn.tex
> ```
>
> I have created a minimal example named llpp_link_minimal_tex.zip,
> please download and unzip it. the tex needed compiled is the
> minimal.tex
>

I don't have tex installed (and would rather not install it if i can
help it)

> ```
> \documentclass{article}
> \usepackage{hyperref}
> \begin{document}
>
> My file links are as follows
>
> \begin{enumerate}
> \item \href{run:local_files/copula.pdf}{relative path pdf link}
> \item
> \href{run:/media/liangzhenjie/Desktop/notes/local_files/llpp_link_minimal_tex/minimal.pdf}{absolute
> path pdf link}
> \item \href{run:local_files/test.tex}{relative path tex link}
> \item
> \href{run:/media/liangzhenjie/Desktop/notes/local_files/llpp_link_minimal_tex/local_files/test.tex}{absolute
> path tex link}
> \end{enumerate}
>
> please change the absolute path in your system to make right. You will
> see that all tex links failed, while all pdf links worked.

Once again lacking any working familiarity with tex the above is greek
to me.

>
>
> \end{document}
>
> ```
>

I guess, the best (only?) way forward is for you to compile your latex
file and share the resulting pdf with me, then i can analyze it for
obvious mistakes, and ask pdf/tex experts for their input.

Bottom line - i will *not* be installing any (la)tex distribution on my
system, sorry.

**Comment by zhenjieliang** [comment]
> I guess, the best (only?) way forward is for you to compile your latex
file and share the resulting pdf with me, then i can analyze it for
obvious mistakes, and ask pdf/tex experts for their input.

Okey, you can download the `llpp_link_minimal_tex.zip` above, unzip and open the `minimal.pdf` file, click the links in the pdf and you will see.

**Comment by moosotc** [comment]
zhenjieliang <notifications@github.com> writes:

>> I guess, the best (only?) way forward is for you to compile your latex
> file and share the resulting pdf with me, then i can analyze it for
> obvious mistakes, and ask pdf/tex experts for their input.
>
> Okey, you can download the `llpp_link_minimal_tex.zip` above, unzip
> and open the `minimal.pdf` file, click the links in the pdf and you
> will see.

Understood. I will discuss it with mupdf guys later today.

**Comment by moosotc** [comment]
zhenjieliang <notifications@github.com> writes:

>> I guess, the best (only?) way forward is for you to compile your latex
> file and share the resulting pdf with me, then i can analyze it for
> obvious mistakes, and ask pdf/tex experts for their input.
>
> Okey, you can download the `llpp_link_minimal_tex.zip` above, unzip
> and open the `minimal.pdf` file, click the links in the pdf and you
> will see.

Okay those the links there are not external actions (or whatever the
proper PDF nomenclature is), but just URI (paths to local files in your
particular subcase), so as instructed llpp tries to open them as such
(i.e. with mupdf machinery and *NOT* via external open (xdg-open and
whatnot))

I'm far into the guess land and rather stop talking now, only mupdf
authors could sched some light on the subject, this is way out of my
league.

**Comment by zhenjieliang** [comment]
> I'm far into the guess land and rather stop talking now, only mupdf
authors could sched some light on the subject, this is way out of my
league.

Your guess is probably correct. I have been using `zathura-pdf-mupdf`, the same error happened when trying to open files from pdf links. After changing `zathura-pdf-mupdf` to `zathura-pdf-poppler` , opening files from pdf links works. 

Maybe the main problem is caused by the mupdf, not other around.

**Comment by zhenjieliang** [comment]
The worst part of zathura is that it cant jump to the correct position from reference link in the pdfs, the best it can do is jump to the correct page. You have to look for the exact equation or figure/tables after jumping to the correct page, which really is a waste of mental energy.

Zathura can be tabbed by the suckless tool [https://tools.suckless.org/tabbed/](tabbed), which is quite useful when you open a lot of pdf files. But a remedy for llpp is rofi search window function.

**Comment by zhenjieliang** [comment]
At the moment llpp works almost perfect for pdf reading, with three functions absence(in my opinion):

(1)combining suckless tabbed, a must for the users opening too many pdf files at the same time. 

(2)opening local files from pdf links, a must for the users who using pdf as their note taking tools.

(3)showing the occurrences of search results.
